### DISCUSSION TITLE
Предварительно загруженные CSS.

### DISCUSSION THREAD

**Discussion by kush1nado** [VKCOM/VKUI#8070] [Q&A] [answered]
Добрый день. 
Столкнулся с такой ситуаций.
Делаю небольшой проект на next.
Я использую компонент Select внутри которого рендерю CustomSelectOption и туда пробрасываю коллекцию изображений.

При получении данных и попытке рендера - происходит следующее:

The resource <URL> was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.

Очень много warnings в консоли подобного типа:
The resource http://localhost:3000/_next/static/css/app/layout.css?v=1734096152395 was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.

Если убираю рендер изображений - warnings уходят.

CSS файлы подгружаются именно из vk-ui.

Подскажите - есть ли какая-то возможность для устранения предупреждения о предварительной загрузке ресурсов?

**Comment by inomdzhon** [comment]
Привет,

Можешь плиз скинуть код вёрстки `Select` + `CustomSelectOption`

**Reply by kush1nado** [reply]
> Привет,
> 
> Можешь плиз скинуть код вёрстки `Select` + `CustomSelectOption`

Конечно. Вот:
```
const {
    field: { onChange, value },
    fieldState: { error },
  } = useController({ name, control });

  const [icons, setIcons] = useState<IconType[] | null>(null);

  useEffect(() => {
    getIcons().then((response) => {
      setIcons(response.data);
    });
  }, []);

  return (
    icons && (
      <FormItem
        top="Иконка"
        htmlFor="icon-id"
        bottom={error?.message && 'Необходимо выбрать иконку'}
        required
      >
        <Select
          value={value ?? ''}
          onChange={onChange}
          id="icon-id"
          placeholder="Не выбран"
          options={icons?.map((icon) => ({
            label: icon.title,
            value: icon.id,
            avatar: icon.src,
          }))}
          renderOption={({ option, ...restProps }) => (
            <CustomSelectOption
              {...restProps}
              key={option.value}
              before={<Avatar src={option.avatar} />}
            />
          )}
        />
      </FormItem>
    )
  );
```

**Reply by kush1nado** [reply]
> before={<Avatar src={option.avatar} />}

Если убираю эту строчку, то warning'и прикращаются

**Reply by inomdzhon** [reply, marked as answer]
Вот [тут](https://stackoverflow.com/questions/73386743/nextjs-image-was-preloaded-using-link-preload-but-not-used-within-a-few-seconds) есть ответ как можно решить проблему, попробуй что-то из того, что рекомендуют
